resolve/index.go

```diff
@@ -110,6 +110,8 @@ type RuleIndex struct {
 
 // ruleRecord contains information about a rule relevant to import indexing.
 type ruleRecord struct {
+	rule  *rule.Rule
```

Comment:
This is still serializable since https://pkg.go.dev/encoding/json#Marshal only acts on exported fields of a structure.

Reply:
The problem will be unmarshalling though, won't it?